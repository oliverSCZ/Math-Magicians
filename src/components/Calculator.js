/* eslint-disable react/prefer-stateless-function */
import React, { Component } from 'react';
import './calculator.css';
import calculate from '../logic/calculate';

class Calculator extends Component {
  constructor(props) {
    super(props);
    this.state = {
      total: 0,
      next: 0,
    };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(e) {
    this.setState((previousState) => calculate(previousState, e.target.innerText));
  }

  render() {
    let val = 0;
    const { total, next } = this.state;
    if (next) {
      val = next;
    } else if (total) {
      val = total;
    }

    return (
      <div className="container">
        <input
          className="grid-item display"
          placeholder="0"
          value={val}
          readOnly
        />
        <button
          type="button"
          className="grid-item AC"
          onClick={this.handleClick}
        >
          AC
        </button>
        <button
          type="button"
          className="grid-item operations"
          onClick={this.handleClick}
        >
          +/-
        </button>
        <button
          type="button"
          className="grid-item procentage"
          onClick={this.handleClick}
        >
          %
        </button>
        <button
          type="button"
          className="grid-item divide operation"
          onClick={this.handleClick}
        >
          รท
        </button>
        <button
          type="button"
          className="grid-item seven"
          onClick={this.handleClick}
        >
          7
        </button>
        <button
          type="button"
          className="grid-item eight"
          onClick={this.handleClick}
        >
          8
        </button>
        <button
          type="button"
          className="grid-item nine"
          onClick={this.handleClick}
        >
          9
        </button>
        <button
          type="button"
          className="grid-item multiply operation"
          onClick={this.handleClick}
        >
          ร
        </button>
        <button
          type="button"
          className="grid-item four"
          onClick={this.handleClick}
        >
          4
        </button>
        <button
          type="button"
          className="grid-item five"
          onClick={this.handleClick}
        >
          5
        </button>
        <button
          type="button"
          className="grid-item six"
          onClick={this.handleClick}
        >
          6
        </button>
        <button
          type="button"
          className="grid-item minus operation"
          onClick={this.handleClick}
        >
          -
        </button>
        <button
          type="button"
          className="grid-item one"
          onClick={this.handleClick}
        >
          1
        </button>
        <button
          type="button"
          className="grid-item two"
          onClick={this.handleClick}
        >
          2
        </button>
        <button
          type="button"
          className="grid-item three"
          onClick={this.handleClick}
        >
          3
        </button>
        <button
          type="button"
          className="grid-item plus operation"
          onClick={this.handleClick}
        >
          +
        </button>
        <button
          type="button"
          className="grid-item zero"
          onClick={this.handleClick}
        >
          0
        </button>
        <button
          type="button"
          className="grid-item period"
          onClick={this.handleClick}
        >
          .
        </button>
        <button
          type="button"
          className="grid-item equal operation"
          onClick={this.handleClick}
        >
          =
        </button>
      </div>
    );
  }
}

export default Calculator;
