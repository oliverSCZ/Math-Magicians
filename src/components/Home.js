import './home.css';

const Home = () => (
  <div className="home">
    <h1>Welcome to our page!</h1>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam sagittis arcu
      nec leo volutpat posuere. Fusce aliquam vel nibh sed sodales. Sed ut augue
      suscipit, suscipit neque id, tempor mauris. In laoreet eleifend ligula in
      rhoncus. Sed ornare rutrum ante, vitae viverra nisl dignissim egestas.
      Nulla facilisi. Quisque sed nunc tincidunt, ullamcorper leo nec, lacinia
      turpis. Quisque quam tortor, accumsan interdum gravida sit amet, hendrerit
      sed augue.
    </p>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam sagittis arcu
      nec leo volutpat posuere. Fusce aliquam vel nibh sed sodales. Sed ut augue
      suscipit, suscipit neque id, tempor mauris. In laoreet eleifend ligula in
      rhoncus. Sed ornare rutrum ante, vitae viverra nisl dignissim egestas.
      Nulla facilisi. Quisque sed nunc tincidunt, ullamcorper leo nec, lacinia
      turpis. Quisque quam tortor, accumsan interdum gravida sit amet, hendrerit
      sed augue.
    </p>
  </div>
);

export default Home;
