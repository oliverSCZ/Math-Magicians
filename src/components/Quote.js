import './quote.css';

const Quote = () => (
  <div className="quote">
    <p>
      Mathematics is not about numbers, equations, computations or algorithms:
      it&apos;s about understanding.
      <br />
      <strong>- William Paul Thurston</strong>
    </p>
  </div>
);

export default Quote;
