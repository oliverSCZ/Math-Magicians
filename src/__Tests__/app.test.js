import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Home from '../components/Home';
import Quote from '../components/Quote';
import Calculator from '../components/Calculator';

describe('Testing components render', () => {
  it('Tests Home.js component render', () => {
    const tree = render(<Home />);
    expect(tree).toMatchSnapshot();
  });
  it('Tests Quote.js component render', () => {
    const tree = render(<Quote />);
    expect(tree).toMatchSnapshot();
  });
  it('Tests Calculator.js component render', () => {
    const tree = render(<Calculator />);
    expect(tree).toMatchSnapshot();
    const output = screen.getByTestId('output');
    userEvent.click(screen.getByText('1'));
    expect(output.value).toContain('1');
    userEvent.click(screen.getByText('+'));
    userEvent.click(screen.getByText('3'));
    userEvent.click(screen.getByText('='));
    expect(output.value).toContain('4');
  });
});
