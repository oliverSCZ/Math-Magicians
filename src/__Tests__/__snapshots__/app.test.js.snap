// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing components render Tests Calculator.js component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="big_container"
      >
        <div
          class="calc_text"
        >
          <h3>
            Let's do some Math!
          </h3>
        </div>
        <div
          class="container"
        >
          <input
            class="grid-item display"
            data-testid="output"
            id="output"
            placeholder="0"
            readonly=""
            value="0"
          />
          <button
            class="grid-item AC"
            type="button"
          >
            AC
          </button>
          <button
            class="grid-item operations"
            type="button"
          >
            +/-
          </button>
          <button
            class="grid-item procentage"
            type="button"
          >
            %
          </button>
          <button
            class="grid-item divide operation"
            type="button"
          >
            ÷
          </button>
          <button
            class="grid-item seven"
            type="button"
          >
            7
          </button>
          <button
            class="grid-item eight"
            type="button"
          >
            8
          </button>
          <button
            class="grid-item nine"
            type="button"
          >
            9
          </button>
          <button
            class="grid-item multiply operation"
            type="button"
          >
            ×
          </button>
          <button
            class="grid-item four"
            type="button"
          >
            4
          </button>
          <button
            class="grid-item five"
            type="button"
          >
            5
          </button>
          <button
            class="grid-item six"
            type="button"
          >
            6
          </button>
          <button
            class="grid-item minus operation"
            type="button"
          >
            -
          </button>
          <button
            class="grid-item one"
            type="button"
          >
            1
          </button>
          <button
            class="grid-item two"
            type="button"
          >
            2
          </button>
          <button
            class="grid-item three"
            type="button"
          >
            3
          </button>
          <button
            class="grid-item plus operation"
            type="button"
          >
            +
          </button>
          <button
            class="grid-item zero"
            type="button"
          >
            0
          </button>
          <button
            class="grid-item period"
            type="button"
          >
            .
          </button>
          <button
            class="grid-item equal operation"
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="big_container"
    >
      <div
        class="calc_text"
      >
        <h3>
          Let's do some Math!
        </h3>
      </div>
      <div
        class="container"
      >
        <input
          class="grid-item display"
          data-testid="output"
          id="output"
          placeholder="0"
          readonly=""
          value="0"
        />
        <button
          class="grid-item AC"
          type="button"
        >
          AC
        </button>
        <button
          class="grid-item operations"
          type="button"
        >
          +/-
        </button>
        <button
          class="grid-item procentage"
          type="button"
        >
          %
        </button>
        <button
          class="grid-item divide operation"
          type="button"
        >
          ÷
        </button>
        <button
          class="grid-item seven"
          type="button"
        >
          7
        </button>
        <button
          class="grid-item eight"
          type="button"
        >
          8
        </button>
        <button
          class="grid-item nine"
          type="button"
        >
          9
        </button>
        <button
          class="grid-item multiply operation"
          type="button"
        >
          ×
        </button>
        <button
          class="grid-item four"
          type="button"
        >
          4
        </button>
        <button
          class="grid-item five"
          type="button"
        >
          5
        </button>
        <button
          class="grid-item six"
          type="button"
        >
          6
        </button>
        <button
          class="grid-item minus operation"
          type="button"
        >
          -
        </button>
        <button
          class="grid-item one"
          type="button"
        >
          1
        </button>
        <button
          class="grid-item two"
          type="button"
        >
          2
        </button>
        <button
          class="grid-item three"
          type="button"
        >
          3
        </button>
        <button
          class="grid-item plus operation"
          type="button"
        >
          +
        </button>
        <button
          class="grid-item zero"
          type="button"
        >
          0
        </button>
        <button
          class="grid-item period"
          type="button"
        >
          .
        </button>
        <button
          class="grid-item equal operation"
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing components render Tests Home.js component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="home"
      >
        <h1>
          Welcome to our page!
        </h1>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam sagittis arcu nec leo volutpat posuere. Fusce aliquam vel nibh sed sodales. Sed ut augue suscipit, suscipit neque id, tempor mauris. In laoreet eleifend ligula in rhoncus. Sed ornare rutrum ante, vitae viverra nisl dignissim egestas. Nulla facilisi. Quisque sed nunc tincidunt, ullamcorper leo nec, lacinia turpis. Quisque quam tortor, accumsan interdum gravida sit amet, hendrerit sed augue.
        </p>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam sagittis arcu nec leo volutpat posuere. Fusce aliquam vel nibh sed sodales. Sed ut augue suscipit, suscipit neque id, tempor mauris. In laoreet eleifend ligula in rhoncus. Sed ornare rutrum ante, vitae viverra nisl dignissim egestas. Nulla facilisi. Quisque sed nunc tincidunt, ullamcorper leo nec, lacinia turpis. Quisque quam tortor, accumsan interdum gravida sit amet, hendrerit sed augue.
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="home"
    >
      <h1>
        Welcome to our page!
      </h1>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam sagittis arcu nec leo volutpat posuere. Fusce aliquam vel nibh sed sodales. Sed ut augue suscipit, suscipit neque id, tempor mauris. In laoreet eleifend ligula in rhoncus. Sed ornare rutrum ante, vitae viverra nisl dignissim egestas. Nulla facilisi. Quisque sed nunc tincidunt, ullamcorper leo nec, lacinia turpis. Quisque quam tortor, accumsan interdum gravida sit amet, hendrerit sed augue.
      </p>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam sagittis arcu nec leo volutpat posuere. Fusce aliquam vel nibh sed sodales. Sed ut augue suscipit, suscipit neque id, tempor mauris. In laoreet eleifend ligula in rhoncus. Sed ornare rutrum ante, vitae viverra nisl dignissim egestas. Nulla facilisi. Quisque sed nunc tincidunt, ullamcorper leo nec, lacinia turpis. Quisque quam tortor, accumsan interdum gravida sit amet, hendrerit sed augue.
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing components render Tests Quote.js component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="quote"
      >
        <p>
          Mathematics is not about numbers, equations, computations or algorithms: it's about understanding.
          <br />
          <strong>
            - William Paul Thurston
          </strong>
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="quote"
    >
      <p>
        Mathematics is not about numbers, equations, computations or algorithms: it's about understanding.
        <br />
        <strong>
          - William Paul Thurston
        </strong>
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
